apps/transport/lib/db/dataset.ex

```diff
@@ -405,6 +390,20 @@ defmodule DB.Dataset do
     end
   end
 
+  def order_datasets(datasets, %{"aom" => aom_id}) do
+    case Integer.parse(aom_id) do
+      {aom_id, ""} ->
+        order_by(datasets,
+          desc: fragment("case when aom_id = ? then 1 else 0 end", ^aom_id),
+          desc: fragment("coalesce(population, 0)"),
+          asc: :custom_title
+        )
+
+      :error ->
+        datasets
+    end
```

Comment:
```suggestion
    aom_id = String.to_integer(aom_id)

    order_by(datasets,
      desc: fragment("case when aom_id = ? then 1 else 0 end", ^aom_id),
      desc: fragment("coalesce(population, 0)"),
      asc: :custom_title
    )
```

Je pense qu'ici aussi on peut se passer de gérer des erreurs de parsing